src/messaging/ExchangeContext.cpp

```diff
@@ -218,7 +218,7 @@ CHIP_ERROR ExchangeContext::SendMessage(Protocols::Id protocolId, uint8_t msgTyp
         }
 #endif
 
-        if (err != CHIP_NO_ERROR && IsResponseExpected())
+        if (err != CHIP_NO_ERROR && sendFlags.Has(SendMessageFlags::kExpectResponse))
```

Comment:
We met an issue when we decreased the tx buffer number on ESP32 and tested it with the homepod mini. After the device receives a Sigma1 message, it sends Sigma2Resume. However the Sigma2Resume sending might fail with the LwIP error ERR_MEM(No TX buffer). Since Sigma2Resume is retransmission-requested, the send will return Success  and the response timer will be started. But the homepod mini won't receive Sigma2Resume if all of the retransmissions fail. So it will send Sigma1 again, then ESP32 will send a Standalone ack back. If this sending fails. the response timer will be canceled. This will cause the device cannot start CASE session establishment again because the sigma1 is unregistered in exchange manager.

Reply:
Added `!IsGroupExchangeContext()`.

Reply:
> then ESP32 will send a Standalone ack back. If this sending fails.

Ok, so yes, this is the "expecting a response, then failing to send a standalone ack in response to a duplicate" case.